packages/components/filters/src/badge/badge.tsx

```diff
@@ -1,5 +1,57 @@
-function Badge() {
-  return <div>badge</div>;
+import { css } from '@emotion/react';
+import { designTokens } from '@commercetools-uikit/design-system';
+
+export type TBadgeProps = {
+  /**
+   * CSS ID for badge, used to specify relationship with parent in parent's `aria-controls` property
+   * see https://www.w3.org/TR/wai-aria-1.1/#status
+   */
+  id: string;
+  /**
+   * If `true`, indicates that the element is in a disabled state.
+   */
+  isDisabled?: boolean;
+  /**
+   * String to be displayed in badge, generally a count of some kind
+   */
+  label: string;
+
+  /**
+   * Optional descriptive explanation of label (e.g. "+4 additional filters applied")
+   */
+  ['aria-label']?: string;
+};
+
+const badgeStyles = css`
```

Comment:
May I propose to use  emotion's `styled` function ([click](https://emotion.sh/docs/styled)) instead of `css` and also use [object styles](https://emotion.sh/docs/object-styles) instead of string literals.

With `styled` you get access to the component props and don't need any additional logic for conditional styles outside of your style-definitions. All styles would be in one `StyledBadge` component.

Additional benefits of the object syntax is, that it is a regular javascript object already understood by your editor and other tools (code quality, refactoring etc).

Reply:
```
const StyledBadge = styled.span({
    fontSize: designTokens.fontSize20,
    fontWeight: designTokens.fontWeight500,
    ... etc...
    // dynamic property based on props
    backgroundColor: (props) =>
        props.isDisabled ? designTokens.colorNeutral : designTokens.colorInfo,
})
```

Reply:
Wow.  I was under the impression that UI Kit heavily favored `css`, but it looks like it uses `styled` just as much.  [Which isnt great](https://emotion.sh/docs/best-practices#use-the-css-prop-or-emotionstyled-but-not-both), but is where we're at.

The one thing that is happening with both `css` and `styled` is that we are _overwhelmingly_ always using the template syntax, and we have linting set up for it.  I personally like the 1-1 template<->css attribute values, but I understand the arguments for object syntax as well.

Also, as far as dynamic styles, the emotion docs [suggest](https://emotion.sh/docs/best-practices#use-the-css-prop-or-emotionstyled-but-not-both) that it is best to use css vars declared in a `style` prop to handle them.

I specifically went with `css` and declaring `css` functions in the array syntax so that we have explicit classes defined for styling certain states, instead of having mostly-similar classes repeated in the DOM with only 1-2 different attrs between each.

All of this is to say that 'wtf do we do about styling' is one of the big maintainability headaches that we need to come to some decisions on, but nothing has been decided yet.

I definitely think that we should at least keep styling consistent _within_ a component, and think I've made a pretty decent case for why we should use `css`, but if we really need to re-evaluate that we can.